`nimsuggest`: failed to build with `clang` on `musl libc`

### What happened?

failed to build `nimsuggest` with `clang` on `musl libc` because of incorrect type of `Thread.sys`

in **nim** `SysThread` defined here:
https://github.com/nim-lang/Nim/blob/v1.6.8/lib/system/threadlocalstorage.nim#L131
in `devel` branch it is defined as `culong`:
https://github.com/nim-lang/Nim/blob/79b25a911be36da87aa012b23e854a3bed01833c/lib/std/private/threadtypes.nim#L75-L76

but in [/usr/include/bits/alltypes.h ](https://git.musl-libc.org/cgit/musl/tree/include/alltypes.h.in#n53) it is defined as struct pointer:
```c
#ifdef __cplusplus
#	if defined(__NEED_pthread_t) && !defined(__DEFINED_pthread_t)
typedef unsigned long pthread_t;
#		define __DEFINED_pthread_t
#	endif

#else

#	if defined(__NEED_pthread_t) && !defined(__DEFINED_pthread_t)
typedef struct __pthread* pthread_t;
#		define __DEFINED_pthread_t
#	endif

#endif

```

### Nim Version

trying to build: `nim-1.6.8`

### Current Standard Output Logs

```shell
CC: nimsuggest.nim
/var/tmp/portage/dev-lang/nim-1.6.8/homedir/.cache/nim/nimsuggest_r/@mnimsuggest.nim.c:4818:21: error:
      incompatible integer to pointer conversion passing 'unsigned long' to parameter of type 'pthread_t'
      (aka 'struct __pthread *') [-Wint-conversion]
        T1_ = pthread_join((*t).sys, ((void**) NIM_NIL));
                           ^~~~~~~~
/usr/include/pthread.h:83:27: note: passing argument to parameter here
int pthread_join(pthread_t, void **);
                          ^
1 error generated.
Error: execution of an external compiler program 'x86_64-gentoo-linux-musl-clang -c -w -fmax-errors=3 -pthread -march=native -O3 -flto -fPIC -falign-functions=32 -funroll-loops -fno-math-errno -fno-trapping-math -fno-math-errno -ffinite-m>


FAILURE
```


### Expected Standard Output Logs

_No response_

### Possible Solution

_No response_

### Additional Information

**_$_** `clang --version`
**clang version**: `15.0.2`
**Target**: `x86_64-gentoo-linux-musl`

----
[musl](https://musl.libc.org/) **version**: `1.2.3`

---

here is fix for [nim 1.6.12](https://github.com/nim-lang/Nim/blob/v1.6.12/lib/system/threadlocalstorage.nim#L131-L132):
```diff
diff --git a/lib/system/threadlocalstorage.nim b/lib/system/threadlocalstorage.nim
--- a/lib/system/threadlocalstorage.nim
+++ b/lib/system/threadlocalstorage.nim
@@ -129,7 +129,7 @@ else:
   when (defined(linux) or defined(nintendoswitch)) and defined(amd64):
     type
       SysThread* {.importc: "pthread_t",
-                  header: "<sys/types.h>" .} = distinct culong
+                  header: "<sys/types.h>" .} = distinct pointer
       Pthread_attr {.importc: "pthread_attr_t",
                     header: "<sys/types.h>".} = object
         abi: array[56 div sizeof(clong), clong]
```
with this change everything compiles fine

i think this should be done somehow with `when` keyword, but i didn't found out how to check if `nim` compiles on `musl` with `when`

-----

when replacing with `object` instead of `pointer`:
```
/var/tmp/portage/dev-lang/nim-1.6.12/homedir/.cache/nim/nimsuggest_r/@m..@slib@ssystem.nim.c:76:26: error:
      typedef redefinition with different types ('struct pthread_t' vs 'struct __pthread *')
typedef struct pthread_t pthread_t;
                         ^
/usr/include/bits/alltypes.h:273:28: note: previous definition is here
typedef struct __pthread * pthread_t;
                           ^
```

--- a/compiler/ccgtypes.nim
+++ b/compiler/ccgtypes.nim
@@ -340,7 +340,12 @@ proc getSimpleTypeDesc(m: BModule; typ: PType): Rope =
   of tyNil: result = typeNameOrLiteral(m, typ, "void*")
   of tyInt..tyUInt64:
     result = typeNameOrLiteral(m, typ, NumericalTypeToStr[typ.kind])
-  of tyDistinct, tyRange, tyOrdinal: result = getSimpleTypeDesc(m, typ.skipModifier)
+  of tyRange, tyOrdinal: result = getSimpleTypeDesc(m, typ.skipModifier)
+  of tyDistinct:
+    result = getSimpleTypeDesc(m, typ.skipModifier)
+    if isImportedType(typ) and result != "":
+      useHeader(m, typ.sym)
+      result = typ.sym.loc.snippet
   of tyStatic:
     if typ.n != nil: result = getSimpleTypeDesc(m, skipModifier typ)
     else:
@@ -861,7 +866,8 @@ proc getTypeDescAux(m: BModule; origTyp: PType, check: var IntSet; kind: TypeDes
   if t != origTyp and origTyp.sym != nil: useHeader(m, origTyp.sym)
   let sig = hashType(origTyp, m.config)
 
-  result = getTypePre(m, t, sig)
+  # tyDistinct matters if it is an importc type
+  result = getTypePre(m, origTyp.skipTypes(irrelevantForBackend-{tyOwned, tyDistinct}), sig)
   defer: # defer is the simplest in this case
     if isImportedType(t) and not m.typeABICache.containsOrIncl(sig):
       addAbiCheck(m, t, result)

--- a/tests/ccgbugs/timportc_distinct.nim
+++ b/tests/ccgbugs/timportc_distinct.nim
@@ -1,0 +1,13 @@
+discard """
+  ccodecheck: "time_t"
+  joinable: false
+"""
+
+type
+  Time* {.importc: "time_t", header: "<time.h>".} = distinct clong
+
+proc foo =
+  var s: Time = default(Time)
+  discard s
+
+foo()

